### ISSUE TITLE
Can't use .fields()

### ISSUE THREAD

**Issue by bolinocroustibat** [collerek/ormar#224]
**Describe the bug**

When I use .fields() method on an existing field, like:

`records = await Comment.objects.filter(criteria=1).fields("id").all()`

or 

`records = await Comment.objects.fields("id").all()`

I get the following error:

```
  File "/Users/myusername/.local/share/virtualenvs/projectAOgmk7mv-py3.9/lib/python3.9/site-packages/ormar/models/newbasemodel.py", line 140, in __init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 2 validation errors for Comment
content
  none is not an allowed value (type=type_error.none.not_allowed)
realness
  none is not an allowed value (type=type_error.none.not_allowed)
```

Is that a bug?

**Versions:**
 - Database backend used: SQlite
 - Python version: 3.9.5
 - `ormar` version: 0.10.9
 - `pydantic` version: 1.8.2
 - `fastapi` version:  0.65.1

**Comment by collerek** [comment]
That's not a bug I wrote you in the #222 you can only do this if other fields are optional.
You cannot exclude a mandatory field (as the name suggest).
So either you need to change the content and realness fields to Optional (i.e. nullable = True, or have to have default value) or you have to include them in fields.

**Comment by bolinocroustibat** [comment]
Gotcha, sorry for the repeat.
If that helps, I got a bit confused by the nullable in Ormar model definition, which, for me, should match the DB constraints, but didn't see how it would constraint the data selection you want rom requests. I thought you would be able to not retrieve a field value in a request, even though it's not nullable in DB.

**Comment by collerek** [comment]
Since you always get a model out of ormar it has to comply to pydantic validation, you cannot have field that is required and populate this model with data without it. Regardless of whether data is coming from dict, Json, database or is declared on instantiation.

You would have to skip validation of the model altogether, which would happen with the enhancement you proposed that you get simply a dict or a list of raw data, but at this point you cannot guarantee that the data fulfill the contract declared by constraints defined by the model. 

Only other option would be to arbitrarily turn off and on the validation of the each field, which not only would be complicated but kind of pointless. Data can be either validated or not, what good comes from partially validated data? It's basically the same as not validated.

**Comment by bolinocroustibat** [comment]
I totally agree that Pydantic models should be used for validation of objects with no exception.
Although, it can happen that you want a constraint/validation on the DB for recording, but you might want to retrieve an object with only the fields that you need. In that case, would that make sense to have the Pydantic models to authorize nullable fields  for the manipulated objects in the code, but to keep the DB constraint IS NOT NULL on the DB level?

**Comment by collerek** [comment]
But that would have to be a different model, the whole point of pydantic (and zen of python) is to fail early.

If you could have a non nullable field in db that can be nullable in ormar you can always save that model and it would fail only on the database call which is kind of late and for example in fastapi it would require custom exception handling instead of validation errors on request (that is performed by fastapi).

So the desired path would be:

*  get the whatever data you want by combining available queryset methods (like select_related, fields etc.)
*  get list /dict instead of ormar model (ehnancement #223)
*  validate returned data with custom pydantic model (NOT ormar model)

But again this validates the data at the end of the process

The other more tricky solution would have to be proxy models, something like described in proxy model inheritance https://collerek.github.io/ormar/models/inheritance/ . In that case that would mean second model referring to the same table, possibly with only a subset of fields, probably read only mode for subset of fields.

**Comment by bolinocroustibat** [comment]
Desired path makes sense. This case would definitely be solved with [#223](https://github.com/collerek/ormar/issues/223).
